### ISSUE TITLE
Lint warning given for external 3rd party package/module reportlab.platypus.flowables

### ISSUE THREAD

**Issue by nanonano** [PyCQA/pylint#8319]
### Bug description

I'm getting a lint warning for a module which is outside of my package, it's `reportlab.platypus.flowables`.

I'm not using `reportlab` directly but it is used by `xhtml2pdf` which I import, see https://github.com/BittyTax/BittyTax/blob/master/bittytax/report.py#L13.

Since I cannot change the code of a 3rd party package, I have tried to suppress the warning using the following but none work.

`ignored-modules=reportlab.platypus.flowables`

`ignore=CVS, reportlab`

You can see the log of the failure here: https://github.com/BittyTax/BittyTax/actions/runs/4227869051/jobs/7342736026

Hope you can help me, I'm completely stuck, it used to work fine in pylint-2.13.9, no warning was produced.

### Configuration

```ini
Not relevant, see https://github.com/BittyTax/BittyTax/blob/master/.pylintrc if required.
```

### Command used

```shell
pylint $(git ls-files '*.py')

See https://github.com/BittyTax/BittyTax/blob/master/.github/workflows/ci.yml.
```

### Pylint output

```shell
************* Module reportlab.platypus.flowables
/opt/hostedtoolcache/Python/3.7.15/x64/lib/python3.7/site-packages/reportlab/platypus/flowables.py:366:12: C0103: Attribute name "firstLineIndent" doesn't conform to snake_case naming style (invalid-name)

------------------------------------
Your code has been rated at 10.00/10
```

### Expected behavior

No warning.

### Pylint version

```shell
pylint-2.16.2
Python/3.7.15

Works okay in pylint-2.13.9 (Python 3.6).
```

### OS / Environment

All Ubuntu, MacOS, Windows.

**Comment by jacobtylerwalls** [comment]
Thanks for the report, I can reproduce.

**Comment by jacobtylerwalls** [comment]
Bisected to a2c57ec06070b6d2485082e3a131a087f8d6d921.

**Comment by nanonano** [comment]
Thanks for the quick response.

Can you think of any workaround to suppress the warning? Apart from `disable=invalid-name`.

**Comment by DanielNoord** [comment]
@jacobtylerwalls How did you reproduce this?

**Comment by jacobtylerwalls** [comment]
I installed BittyTax in editable mode like the linked CI job.

**Comment by DanielNoord** [comment]
I did a little investigation and for some reason the `instance_attrs` of the `Buy` `ClassDef` gets populated with stuff that shouldn't be here.

I suspect the issue is with `pylint $(git ls-files '*.py')`. I tried just doing `pylint src` and I got no errors. Why don't you use that invocation?

**Comment by DanielNoord** [comment]
The issue is not in that file, but in `transaction.py`. That class gets its instance attributes mangled up, probably because we don't handle the caching correctly.

I would suggest using `pylint src --recursive=y` for now as this is quite hard to debug and might not get resolved very quickly.
